### PR TITLE
Add bundling support

### PR DESCRIPTION
**Pull request by atulsriv**
- Add helper utils for creating tip instruction
- Add helper utils for creating tip transaction
- Add support in providers to support modified endpoints
-

### CODE REVIEW

File: provider/ws.go [hunk 1]
```diff
@@ -393,6 +397,35 @@ func (w *WSClient) PostSubmit(ctx context.Context, txBase64 string, skipPreFligh
 	return &response, nil
 }
 
+// PostSubmitJitoBundle takes in a list of transactions, signs them, and submits them to PostSubmitJitoEndpoint
```
Comment (DimaCantemir) [inline provider/ws.go]: either uncomment or remove

File: provider/http.go [hunk 2]
```diff
@@ -329,6 +331,28 @@ func (h *HTTPClient) PostSubmit(ctx context.Context, txBase64 string, skipPreFli
...
+// PostSubmitWithBundle signs a transaction and submits it using front running protection
+func (h *HTTPClient) PostSubmitWithBundle(ctx context.Context, txBase64 string) (*pb.PostSubmitResponse, error) {
+	url := fmt.Sprintf("%s/api/v2/submit", h.baseURL)
+
+	signedTx, err := transaction.SignTxWithPrivateKey(txBase64, *h.privateKey)
+	if err != nil {
+		return &pb.PostSubmitResponse{}, err
+	}
+
+	frontRunningProtection := true
```
Comment (DimaCantemir) [inline provider/http.go]: Why is the hardcoded ?

**Review by DimaCantemir — CHANGES_REQUESTED** [review]
please cleanup and make consistent changes across all providers

File: provider/http.go (hunk 2 above)
Comment (atulsriv) [reply provider/http.go]: removed

File: provider/ws.go [hunk 3]
```diff
@@ -404,15 +408,24 @@ func (w *WSClient) PostSubmitBatch(ctx context.Context, request *pb.PostSubmitBa
 }
 
 // PostSubmitV2 posts the transaction string to the Solana network.
-func (w *WSClient) PostSubmitV2(ctx context.Context, txBase64 string, skipPreFlight bool) (*pb.PostSubmitResponse, error) {
+func (w *WSClient) PostSubmitV2(ctx context.Context, txBase64 string, skipPreFlight bool, useBundle bool) (*pb.PostSubmitResponse, error) {
+	if w.privateKey == nil {
+		return &pb.PostSubmitResponse{}, ErrPrivateKeyNotFound
+	}
+
+	txBase64, err := transaction.SignTxWithPrivateKey(txBase64, *w.privateKey)
```
Comment (DimaCantemir) [inline provider/ws.go:426]: why don't we have this in the other providers ?

**Review by DimaCantemir — APPROVED** [review]

**Review by DimaCantemir — APPROVED** [review]

**Pull request merged**